src/ram_forest.cpp

```diff
@@ -531,5 +531,63 @@ bool RamForest::operator==(const RamForest& other)
            roots == other_roots &&
            m_posmap == other.m_posmap;
 }
+std::string RamForest::tostring(const ForestState& f)
+{
+    uint8_t fh = f.NumRows();
+    if (fh > 6) {
+        std::string s = "can't Print" + std::to_string(f.m_num_leaves) + "Leaves. Roots:\n";
+        std::vector<utreexo::Hash> roots;
+        Roots(roots);
+        for (int i = 0; i < roots.size(); i++) {
+            s += "\t" + std::to_string(i);
+            for(int j=0;j<12;j++)
+            {
+                s+=std::to_string(roots[i][j]);
+            }
+            s+='\n';
+        }
+        return s;
+    }
+    std::vector<std::string> output((fh * 2) + 1, "");
+    uint8_t pos;
+    for (int h = 0; h <= fh; h++) {
+        unsigned int rowlen = (1 << (fh - h));
```

Comment:
I try to use the coding style recommend in the [Bitcoin Core developer docs](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#coding-style-c).
```suggestion
        unsigned int row_length = (1 << (fh - h));
```

Reply:
I made all the changes that you requested. Thank you.